src/software/geom/spline.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include <vector>
+
+#include "software/geom/point.h"
+#include "software/geom/polynomial.h"
+
+/**
+ * A Spline maps a double to a Point with a parameterized piecewise function
+ * composed of Polynomials over ranges
+ */
+class Spline
+{
+   public:
+    Spline() = delete;
+    /**
+     * Construct a spline by drawing line segments between consecutive
+     * Points
+     *
+     * @throws std::runtime_error if points.size() == 1
```

Comment:
I think size of 1 should be valid (every value of `t` just maps to the same point), but an empty list should not be valid here (what does `t` map to?).

Reply:
are we gonna go back to std::optional to return no path again?

Reply:
That would make sense, that's our general pattern across the codebase for if we don't have something to return.

Reply:
done